lib/workers/repository/process/write.ts

```diff
@@ -42,15 +42,17 @@ export async function writeUpdates(
   for (const branch of branches) {
     addMeta({ branch: branch.branchName });
     const branchExisted = branchExists(branch.branchName);
-    let branchCache = cachedBranches?.find(
-      (br) => br.branchName === branch.branchName
-    );
+    let branchCache = {} as BranchCache;
+    if (cachedBranches.length) {
```

Comment:
```suggestion
    if (branchExisted && cachedBranches.length) {
```

Shouldn't exist/be usable if the branch is not present?

Reply:
updated to check branhExists first and then try to find its cache